resources/oti/queries.sql

```diff
@@ -187,10 +187,9 @@ SELECT r.state, r.retry, p.type AS payment_type
 LEFT JOIN payment p ON r.id = p.registration_id
 WHERE r.participant_id = :id;
 
--- name: insert-participant!
-INSERT INTO participant (ext_reference_id, email, language_code)
-  SELECT :external-user-id, :email, :language-code
-  WHERE NOT EXISTS (SELECT id FROM participant WHERE ext_reference_id = :external-user-id);
+-- name: upsert-participant!
+INSERT INTO participant (ext_reference_id, email, language_code) VALUES (:external-user-id, :email, :language-code)
```

Comment:
Miksi tämä on, onko tämä siis siksi että aina viimeisin ilmoittautuminen määrittää tutkintotodistuksen kielen?

Reply:
Joo, se on toiminut näin aikaisemminkin
https://github.com/Opetushallitus/oti/blob/724957229ccb6d60a3cd2f9b275af49623d2e72f/src/clj/oti/service/user_data.clj#L170
`(->> db-data (sort-by :registration_id) last :registration_language)`